cdc/sink/common/common.go

```diff
@@ -88,40 +117,44 @@ func (c *UnresolvedTxnCache) Resolved(resolvedTs uint64) map[model.TableID][]*mo
 	return resolvedTxnsMap
 }
 
-// Unresolved returns unresolved txns
-func (c *UnresolvedTxnCache) Unresolved() map[model.TableID][]*model.SingleTableTxn {
-	return c.unresolvedTxns
-}
-
 // UpdateCheckpoint updates the checkpoint ts
 func (c *UnresolvedTxnCache) UpdateCheckpoint(checkpointTs uint64) {
 	atomic.StoreUint64(&c.checkpointTs, checkpointTs)
 }
 
 func splitResolvedTxn(
-	resolvedTs uint64, unresolvedTxns map[model.TableID][]*model.SingleTableTxn,
+	resolvedTs uint64, unresolvedTxns map[model.TableID][]txnsWithTheSameCommitTs,
 ) (minTs uint64, resolvedRowsMap map[model.TableID][]*model.SingleTableTxn) {
 	resolvedRowsMap = make(map[model.TableID][]*model.SingleTableTxn, len(unresolvedTxns))
 	minTs = resolvedTs
 	for tableID, txns := range unresolvedTxns {
 		i := sort.Search(len(txns), func(i int) bool {
-			return txns[i].CommitTs > resolvedTs
+			return txns[i].commitTs > resolvedTs
 		})
 		if i == 0 {
 			continue
 		}
-		var resolvedTxns []*model.SingleTableTxn
+		var resolvedTxnsWithTheSameCommitTs []txnsWithTheSameCommitTs
 		if i == len(txns) {
-			resolvedTxns = txns
+			resolvedTxnsWithTheSameCommitTs = txns
 			delete(unresolvedTxns, tableID)
 		} else {
-			resolvedTxns = txns[:i]
+			resolvedTxnsWithTheSameCommitTs = txns[:i]
 			unresolvedTxns[tableID] = txns[i:]
 		}
```

Comment:
`unresolvedTxns` holds a slice of struct  `txnsWithTheSameCommitTs`, copy could happen easily and have performance degradation. Suggest to use struct pointer in slice, or we can have a benchmark